vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/multiple_string.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {useState, useMemo, useEffect, useCallback, useRef} from 'react';
+import * as _ from 'lodash';
+
+import {InputBackendComponent} from './input_types';
+import TextInput from 'etna-js/components/inputs/text_input';
+
+function equalKeys(hash1: {}, hash2: {}): boolean {
+  return _.isEqual(Object.keys(hash1), Object.keys(hash2));
+}
+
+const MultipleStringInput: InputBackendComponent = ({
+  input,
+  onChange
+}) => {
+  const {data, name} = input;
+
+  const options: {[k: string]: string} = useMemo(() => {
+    if (data) {
+      return Object.values(data).reduce((a, b) => ({...a, ...b}), {});
+    };
+
+    return {};
+  }, [data]);
+
+  useEffect(() => {
+    // Set all key's values as the initially given values, only if no input.value provided. (Initialization)
+    // Otherwise set to mirror input.value
+    if (null == input.value) {
+      onChange(name, options);
+    }
+  }, [options, input.value, name, onChange]);
+
+  const updateLabel = (newValue: string, key: string, prevLabels = input.value) => {
+    prevLabels[key] = newValue;
+    onChange(name, prevLabels);
+    console.log(input.value);
```

Comment:
This console.log only seems to be called once or twice per keystroke, so seems there's little recursion introduced!

The use of input.value rather than direct state seems to have been enough =)

Reply:
(I do plan to remove this before merge.)